File: add/codesnippet/CSharp/p-system.windows.forms.r_0_1.cs
      private void ProtectMySelectedText()
      {
         // Determine if the selected text in the control contains the word "RichTextBox".
         if(richTextBox1.SelectedText != "RichTextBox")
         {
            // Search for the word RichTextBox in the control.
            if(richTextBox1.Find("RichTextBox",RichTextBoxFinds.WholeWord)== -1)
            {
               //Alert the user that the word was not foun and return.
               MessageBox.Show("The text \"RichTextBox\" was not found!");
               return;
            }
         }
         // Protect the selected text in the control from being altered.
         richTextBox1.SelectionProtected = true;
      }

File: add/codesnippet/VisualBasic/p-system.windows.forms.r_0_1.vb
   Private Sub ProtectMySelectedText()
      ' Determine if the selected text in the control contains the word "RichTextBox".
      If richTextBox1.SelectedText <> "RichTextBox" Then
         ' Search for the word RichTextBox in the control.
         If richTextBox1.Find("RichTextBox", RichTextBoxFinds.WholeWord) = -1 Then
            'Alert the user that the word was not foun and return.
            MessageBox.Show("The text ""RichTextBox"" was not found!")
            Return
         End If
      End If
      ' Protect the selected text in the control from being altered.
      richTextBox1.SelectionProtected = True
   End Sub